clients/src/main/java/org/apache/kafka/clients/tools/VerifiableProducer.java

```diff
@@ -150,8 +174,24 @@ public static VerifiableProducer createFromArgs(String[] args) {
             producerProps.put(ProducerConfig.ACKS_CONFIG, Integer.toString(res.getInt("acks")));
             // No producer retries
             producerProps.put("retries", "0");
+            
+            // Properties specified using the --producer-prop key=val format override
+            // other properties
+            List<String> userSpecifiedProducerProps = res.getList("userSpecifiedProducerProps");
+            for (String prop: userSpecifiedProducerProps) {
```

Comment:
Nothing like this in clients/tools, since this is the first introduction of an option parsing library,

There may be something analogous in core.

However I'm not sure if it's good to add a dependency on core, where jopt-simple is used. Also, as noted with the problems parsing "-1", jopt-simple had some impressively obnoxious corner cases.